### DISCUSSION TITLE
What is Clarity's stance with regards to custom styling?

### DISCUSSION THREAD

**Discussion by alehar9320** [vmware-clarity/core#66] [Q&A] [answered]
Hi!

Great work on creating the Clarity Design System. I'm intrigued to possibly use the Clarity Design System, but I sense that the ability to customize the styling is quite limited. 

What is Clarity's stance with regards to custom styling? Is the long-term goal to be close to the VMWare brand or is there a more generic approach?

Thanks.

**Comment by ashleyryan** [comment, marked as answer]
Just about everything in Clarity Core is tokenized, making it easier to create a custom theme for your use. We focus heavily on using css variables related to interactions, instead of referencing color names, so that updating a background color will update it everything.

https://clarity.design/storybook/core/?path=/story/foundation-design-tokens--page
https://clarity.design/storybook/core/?path=/story/themes-getting-started--page

Our dark mode stylesheet, for example, simply changes the values of some of the CSS variables: https://unpkg.com/@cds/core@6.0.0/styles/theme.dark.css

The individual components themselves also expose css variables that you can tweak to meet your need.  Our stance is that you may use the CSS variables however you wish, and limit targeting the elements themselves in your CSS selectors (which the Shadow DOM makes impossible to do in many cases anyway). Limiting customizations to the tokens means easier upgrades in the future.

**Reply by colinreedmiller** [reply]
A crucial aspect to the answer to your question is: Clarity NG or Clarity Core (web components)?
NG has more limited theming, Core development is currently paused.

**Reply by TekSiDoT** [reply]
> Core development is currently paused.

Wait what? We're heavily invested in Clarity and are currently transitioning to Core. I have found no mention of this anywhere else on either the different repos or Clarity websites, would you mind to elaborate?

**Reply by ashleyryan** [reply]
Our current priority is tackling the massive backlog of Accessibility issues (from the VPAT audit) related to Clarity NG.  If you're a vmware employee, we've announced this on slack, but you're right that we haven't put anything on the websites or repos.

**Reply by TekSiDoT** [reply]
Thank you Ashley.

As a long time Clarity user and internal advocate of design systems in general I must admit that I'm a little worried about Claritys future:

While the move to Web Components makes absolute sense (Angular is no longer the default choice for many corporate teams), it necessarily meant a halt on feature development for Angular specific content. It also feels like the transition to Core is a lot slower than anticipated.

Change in team management, mentions of "If you're a vmware employee" or "contact Person xyc via mail for product plan updates" make me doubt the long term open-source investment of vmware into this project. Also, while I appreciate accessibility is a large issue, dropping everything to focus development on this seems ... weird.

I apologize if this post seems to be overly negative. I've greatly enjoyed using Clarity for several years now and am very grateful for the work you have done for the community. Alas, from a business point of view I need to be certain that our continued investment in migrating to @cds/core is justified and the Clarity Design System does not indeed have an uncertain vmware internal future in some kind of maintenance mode for existing projects.

**Reply by ashleyryan** [reply]
I understand completely your frustration.

A year or two ago, vmware underwent a massive accessibility audit of its products. It became clear that Clarity had a huge number of accessibility issues that were impacting the accessibility of the company's products. I can understand why it may seem weird to drop everything and focus on accessibility, the reality is that this is a huge priority for the company right now and as such we've had no choice but to focus on it. 

For the sake of transparency - the Clarity team underwent a change of leadership in January, following that a good amount of attrition. We have not done a good job of messaging around these changes, and for that, I apologize. There has been a lot of conversation and discussion around the future and direction of Clarity. Here is what I know for now - currently the team's focus is on accessibility, particularly it's angular components. Similarly, the team is placing more emphasis on maintenance and support of Clarity NG. We've realized that adoption and transition to Core has been a lot slower than anticipated, like you said. 

I can't tell you right now what the exact future or roadmap of Core is. At this moment, we remain committed to supporting it for our users. There are many products at vmware that don't use Angular. Clarity is the design system for VMware - it's critical to the goal of a One VMware and isn't going anywhere.

**Reply by TekSiDoT** [reply]
Thank you again for your reply.

Are there any update on the project and roadmap state? 

We've currently stopped all work related to Clarity Core and have begun replacing Clarity components in our existing codebase whenever we have the chance.

**Reply by TekSiDoT** [reply]
Is there any update on this issue?
